spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/system/ExecutorInstrumentationImplementation.java

```diff
@@ -0,0 +1,24 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.spi.system;
+
+/**
+ * Interface for custom implementations of {@link ExecutorInstrumentation}.
+ * <p>
+ * A distinct interface is needed for dependency injection to resolve the optional
+ * {@link ExecutorInstrumentationImplementation} vs. the mandatory {@link ExecutorInstrumentation}
+ * that is provided by core services, providing a default implementation if none is provided.
```

Comment:
I don't understand why the control is better with additional optional `ExecutorInstrumentationImplementation` interface. Could you give some example to see what's the benefit from it?

Reply:
Micrometer extension provides a `ExecutorInstrumentationImplementation`.
Core extension consumes an (optional) `ExecutorInstrumentationImplementation` and registers a `ExecutorInstrumentation` (with either the `ExecutorInstrumentationImplementation` instance or a default implementation).
Downstream modules consume a (mandatory) `ExecutorInstrumentation`.

Because of the way dependencies are resolved (with a graph) it's not possible for Core extension to consume and register the same service.

Reply:
thanks for explanation!

Reply:
Added to javadoc